### ISSUE TITLE
custom_mask grows unbounded via torch.cat in EAGLE/DFlash speculative decoding

### ISSUE THREAD

**Issue by ItzDEXX** [sgl-project/sglang#37553]
### Summary

`EagleVerifyInput.generate_attn_arg_prefill` and `DFlashVerifyInput.generate_attn_arg_prefill` grow the `custom_mask` tensor via `torch.cat` on every batch-size increase and store the result back on `self`. The mask never shrinks, causing unbounded GPU memory growth and stale data returned to the attention kernel.

### Affected files

- `python/sglang/srt/speculative/eagle_info.py:123-136`
- `python/sglang/srt/speculative/dflash_info.py:190-204`

### Root cause

The `custom_mask` tensor is reused across decode steps (stored on `self`). When the batch size increases, `torch.cat` appends `True` padding to make the mask large enough:

```python
self.custom_mask = torch.cat(
    [self.custom_mask, torch.full((mask_numel - self.custom_mask.numel(),), True, ...)],
    dim=0,
)
```

Two problems:

1. **Unbounded memory growth** — `torch.cat` allocates a new tensor every time and the old one becomes garbage. The mask only grows, never shrinks. Over a long-running serving session with fluctuating batch sizes, it accumulates to peak size and stays there, churning the CUDA allocator.

2. **Stale data returned** — when the batch shrinks, the function returns the full oversized `self.custom_mask` (not a slice), so the attention kernel receives more elements than `mask_numel`, including stale `True` padding from a previous larger batch.

The FIXME comments at both sites confirm this was known to be a temporary fix:
```
# FIXME(attn): temporary fix for custom mask padding with cuda graph
```

### Impact

- Slow GPU memory leak in long-running serving workloads with variable batch sizes
- Potential OOM in extreme cases due to allocator fragmentation
- Attention kernel receives stale padding data when batch size decreases

### Environment

- SGLang main (as of 2026-08-30)
- Any GPU, any model using EAGLE or DFlash speculative decoding

### Fix

PR #37120 replaces `torch.cat` with a fresh `torch.full` allocation at exactly `mask_numel` and returns a `[:mask_numel]` slice when the mask is already large enough.
